### ISSUE TITLE
python: segfault when accessing `description` after fetching rows

### ISSUE THREAD

**Issue by cldellow** [duckdb/duckdb#6315]
### What happens?

Python code that worked in 0.6.1 segfaults in 0.7.0.

### To Reproduce

```
python3 -mvenv venv
. venv/bin/activate
pip install duckdb==0.7.0
python3
>>> import duckdb
>>> c = duckdb.connect()
>>> rv = c.execute("select * from sqlite_master where type = 'table'")
>>> rv.fetchall()
[]
>>> rv.description
Segmentation fault (core dumped)
```

Contrasting with 0.6.1:

```
>>> import duckdb
>>> c = duckdb.connect()
>>> rv = c.execute("select * from sqlite_master where type = 'table'")
>>> rv.fetchall()
[]
>>> rv.description
[('type', 'STRING', None, None, None, None, None), ('name', 'STRING', None, None, None, None, None), ('tbl_name', 'STRING', None, None, None, None, None), ('rootpage', 'NUMBER', None, None, None, None, None), ('sql', 'STRING', None, None, None, None, None)]
```

### OS:

Ubuntu 20.04.3, x64

### DuckDB Version:

0.7.0

### DuckDB Client:

Python

### Full Name:

Colin Dellow

### Affiliation:

Hobbyist

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Mause** [comment]
Does this only happen when querying the sqlite_master view?

**Comment by cldellow** [comment]
Hm, good question. Looks like it happens for simpler queries:

```
>>> import duckdb
>>> c = duckdb.connect()
>>> rv = c.execute('select 1')
>>> rv.fetchall()
[(1,)]
>>> rv.description
Segmentation fault (core dumped)
```

**Comment by cldellow** [comment]
Ah, but I can access the description before I retrieve the rows:

```
>>> import duckdb
>>> c = duckdb.connect()
>>> rv = c.execute('select 1')
>>> rv.description
[('1', 'NUMBER', None, None, None, None, None)]
>>> rv.fetchall()
[(1,)]
>>> rv.description
Segmentation fault (core dumped)
```

**Comment by Mause** [comment]
I guess that explains how it wasn't caught before? In most cases you'd get the description first

**Comment by cldellow** [comment]
Ah, could be! I'm not very familiar with the DB API beat practices. I was translating some stuff from SQLite to DuckDB and just fiddled away until it worked.

**Comment by Mytherin** [comment]
Thanks for the report! I have pushed a fix in #6326

**Comment by cldellow** [comment]
Awesome, thank you!
